### ISSUE TITLE
Container Resource Request defaulting to Resource Limit

### ISSUE THREAD

**Issue by alextadams88** [kubernetes/kubernetes#117372]
### What happened?

I'm working with Kubernetes and I want to make a deployment with a resource limit to ensure that it never goes over a certain amount of usage. But I don't need to set any resource reservations, it can go as little as possible. So I create a manifest without specifying `spec.template.spec.containers[*].resources.requests` but, setting values for `spec.template.spec.containers[*].resources.limits`.

After deploying this manifest, I notice that my container is keeping reserved the memory set equal to `spec.template.spec.containers[*].resources.limits`. This was unexpected behavior, I would expect memory usage to be simply whatever the container's needs are.

### What did you expect to happen?

I would expect that any container manifest that does not specify any `spec.template.spec.containers[*].resources.requests` would not reserve any resources. Maximum resource limits should have no impact on minimum resource reservations. 

TL;DR: If I set a _maximum_ value for my resource limits, but no minimum, then I would never expect for the minimum to be automatically set equal to the maximum. 

### How can we reproduce it (as minimally and precisely as possible)?

Deploy a manifest file with no `spec.template.spec.containers[*].resources.requests` but with `spec.template.spec.containers[*].resources.limits`.

Observe the manifest file after deployment, and observe that `spec.template.spec.containers[*].resources.requests` has the same values as `spec.template.spec.containers[*].resources.limits`

### Anything else we need to know?

I understand that kubernetes documentation states that this is the expected behavior:
https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-default-namespace/#what-if-you-specify-a-container-s-limit-but-not-its-request

But I just can't understand why this behavior would be desired. I guess I am creating this issue just in the hopes of understanding _why_, and wondering if this behavior can be changed. To me, although it is expected and documented, it is counter-intuitive and can cause users to inadvertently have high minimum resource reservations for their containers. 

### Kubernetes version

<details>
1.25.5
</details>

### Cloud provider

<details>
Azure
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by tzneal** [comment]
This is expected behavior, see the docs at https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits. Specifically:

> Note: If you specify a limit for a resource, but do not specify any request, and no admission-time mechanism has applied a default request for that resource, then Kubernetes copies the limit you specified and uses it as the requested value for the resource.

**Comment by alextadams88** [comment]
Right, but my question is _why_ is it expected and I am proposing that it be changed. To me it seems very counter intuitive. For example, if I specify a maximum memory value of 2 GB for a container, the _last_ thing I want is for that container to keep 2 GB in reserve at all times, but that's what will happen.

**Comment by SergeyKanzhelev** [comment]
/remove-kind bug
/kind feature

@alextadams88 what is the scenario so you don't want to set a request?
